examples/screen_cast_pw.rs

```diff
@@ -0,0 +1,218 @@
+            if id != pw::spa::param::ParamType::Format.as_raw() {
+                return;
+            }
+
+            let (media_type, media_subtype) =
+                match pw::spa::param::format_utils::parse_format(param) {
+                    Ok(v) => v,
+                    Err(_) => return,
+                };
+
+            if media_type != pw::spa::param::format::MediaType::Video
+                || media_subtype != pw::spa::param::format::MediaSubtype::Raw
+            {
+                return;
+            }
+
+            user_data
+                .format
+                .parse(param)
+                .expect("Failed to parse param changed to VideoInfoRaw");
+
+            println!("got video format:");
+            println!(
+                "  format: {} ({:?})",
+                user_data.format.format().as_raw(),
+                user_data.format.format()
+            );
+            println!(
+                "  size: {}x{}",
+                user_data.format.size().width,
+                user_data.format.size().height
+            );
+            println!(
+                "  framerate: {}/{}",
+                user_data.format.framerate().num,
+                user_data.format.framerate().denom
+            );
+
+            // prepare to render video of this size
+        })
+        .process(|stream, _| {
+            match stream.dequeue_buffer() {
+                None => println!("out of buffers"),
+                Some(mut buffer) => {
+                    let datas = buffer.datas_mut();
+                    if datas.is_empty() {
+                        return;
+                    }
+
+                    // copy frame data to screen
+                    let data = &mut datas[0];
+                    println!("got a frame of size {}", data.chunk().size());
+                }
+            }
+        })
+        .register()?;
+
+    println!("Created stream {:#?}", stream);
+
+    let obj = pw::spa::pod::object!(
+        pw::spa::utils::SpaTypes::ObjectParamFormat,
+        pw::spa::param::ParamType::EnumFormat,
+        pw::spa::pod::property!(
+            pw::spa::param::format::FormatProperties::MediaType,
+            Id,
+            pw::spa::param::format::MediaType::Video
+        ),
+        pw::spa::pod::property!(
+            pw::spa::param::format::FormatProperties::MediaSubtype,
+            Id,
+            pw::spa::param::format::MediaSubtype::Raw
+        ),
+        pw::spa::pod::property!(
+            pw::spa::param::format::FormatProperties::VideoFormat,
+            Choice,
+            Enum,
+            Id,
+            pw::spa::param::video::VideoFormat::RGB,
+            pw::spa::param::video::VideoFormat::RGB,
+            pw::spa::param::video::VideoFormat::RGBA,
+            pw::spa::param::video::VideoFormat::RGBx,
+            pw::spa::param::video::VideoFormat::BGRx,
+            pw::spa::param::video::VideoFormat::YUY2,
+            pw::spa::param::video::VideoFormat::I420,
```

Comment:
I'd suggest to have all 8 32-bit rgb formats (https://gitlab.freedesktop.org/pipewire/pipewire/-/blob/master/spa/include/spa/param/video/raw.h?ref_type=heads#L45-52) in the list here so compositors/drivers can choose the optimal one.

Also the duplicated `RGB` at the start means that it`s the preferred format - which is not great as 24-bit RGB can be really slow on modern GPUs. I'd suggest to drop `RGB` altogether - or at least but some 32-bit format at the top.

Reply:
Thanks for your input, this is quite insightful! @rmader 

How about this sequence then? I've de-prioritised RGB based formats to death.

```
pw::spa::param::video::VideoFormat::BGRx,
pw::spa::param::video::VideoFormat::BGRx,
pw::spa::param::video::VideoFormat::xBGR,
pw::spa::param::video::VideoFormat::BGRA,
pw::spa::param::video::VideoFormat::ABGR,
pw::spa::param::video::VideoFormat::YUY2,
pw::spa::param::video::VideoFormat::I420,
pw::spa::param::video::VideoFormat::RGBA,
pw::spa::param::video::VideoFormat::ARGB,
pw::spa::param::video::VideoFormat::xRGB,
pw::spa::param::video::VideoFormat::RGBx,
pw::spa::param::video::VideoFormat::RGBA,
pw::spa::param::video::VideoFormat::RGBx,
pw::spa::param::video::VideoFormat::RGB,
```

Reply:
Only the first format should matter - that's the default pick (it has to be listed again as option afterwards). The rest is up to you, but it might make sense to follow some order, such as:

```
pw::spa::param::video::VideoFormat::BGRx,
pw::spa::param::video::VideoFormat::ABGR,
pw::spa::param::video::VideoFormat::xBGR,
pw::spa::param::video::VideoFormat::ARGB,
pw::spa::param::video::VideoFormat::xRGB,
pw::spa::param::video::VideoFormat::BGRA,
pw::spa::param::video::VideoFormat::BGRx,
pw::spa::param::video::VideoFormat::RGBA,
pw::spa::param::video::VideoFormat::RGBx,
pw::spa::param::video::VideoFormat::RGB,
pw::spa::param::video::VideoFormat::I420,
pw::spa::param::video::VideoFormat::YUY2,
```

Reply:
we can keep BGRx as the default pick, yeah?

Reply:
Let us land this as is, if default format could be improved, it can be done in a separate PR.